packages/flutter_tools/lib/src/base/analyze_size.dart

```diff
@@ -165,52 +200,76 @@ class SizeAnalyzer {
     _SymbolNode currentNode,
     String totalPath,
     _SymbolNode aotSnapshotJsonRoot,
+    int maxDepth,
+    int currentDepth,
   ) {
     totalPath += currentNode.name;
 
     assert(_appFilename != null);
     if (currentNode.children.isNotEmpty
-      && currentNode.name != '$_appFilename (Dart AOT)') {
+      && currentNode.name != '$_appFilename (Dart AOT)'
+      && currentDepth < maxDepth
+      && currentNode.byteSize >= 1000) {
       for (final _SymbolNode child in currentNode.children) {
-        _printLibChildrenPaths(child, '$totalPath/', aotSnapshotJsonRoot);
+        _printLibChildrenPaths(child, '$totalPath/', aotSnapshotJsonRoot, maxDepth, currentDepth + 1);
       }
+      _leadingPaths = totalPath.split('/')
+        ..removeLast();
     } else {
-      // Print total path and size if currentNode does not have any chilren.
-      _printEntitySize(totalPath, byteSize: currentNode.byteSize, level: 2);
-
-      // We picked this file because arm64-v8a is likely the most popular
-      // architecture. Other architecture sizes should be similar.
-      final String libappPath = 'lib/arm64-v8a/$_appFilename';
-      // TODO(peterdjlee): Analyze aot size for all platforms.
-      if (totalPath.contains(libappPath)) {
-        _printAotSnapshotSummary(aotSnapshotJsonRoot);
+      // Print total path and size if currentNode does not have any children and is
+      // larger than 1KB
+      final bool isAotSnapshotPath = _locatedAotFilePath.join('/').contains(totalPath);
+      if (currentNode.byteSize >= 1000 || isAotSnapshotPath) {
+        _printEntitySize(totalPath, byteSize: currentNode.byteSize, level: 1, emphasis: currentNode.children.isNotEmpty);
+        if (isAotSnapshotPath) {
+          _printAotSnapshotSummary(aotSnapshotJsonRoot, level: totalPath.split('/').length);
+        }
+        _leadingPaths = totalPath.split('/')
+          ..removeLast();
       }
     }
   }
 
   /// Go through the AOT gen snapshot size JSON and print out a collapsed summary
   /// for the first package level.
-  void _printAotSnapshotSummary(_SymbolNode aotSnapshotRoot, {int maxDirectoriesShown = 10}) {
+  void _printAotSnapshotSummary(_SymbolNode aotSnapshotRoot, {int maxDirectoriesShown = 30, @required int level}) {
```

Comment:
I think 10 is probably fine (it's more or less a screenful) since lower sized packages are a diminishing return as we show them. There's always DevTools :)

Reply:
> I think 10 is probably fine (it's more or less a screenful) since lower sized packages are a diminishing return as we show them. There's always DevTools :)

showing 10 is pointless, it will be almost identical for every single flutter using package. At this point we have removed 95 % of the useful console output, we might as well skip it all together.

Reply:
discussed on main thread